Sources/Controllers/Settings/WidgetSettings/WidgetGroupListViewController.swift

```diff
@@ -168,27 +168,54 @@ class WidgetGroupListViewController: OABaseNavbarViewController, UISearchBarDele
     }
 
     func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        if searchText.isEmpty {
-            isFiltered = false
-            setupSearchControllerWithFilter(false)
-        } else {
-            isFiltered = true
-            setupSearchControllerWithFilter(true)
-            filteredSection.removeAllRows()
-            let section = tableData.sectionData(for: 0)
-            for i in 0..<section.rowCount() {
-                let row = section.getRow(i)
-                let name = row.title
-                guard let name else { continue }
-                let nameRange = name.range(of: searchText, options: .caseInsensitive)
-                if nameRange != nil {
+        isFiltered = !searchText.isEmpty
+        setupSearchControllerWithFilter(isFiltered)
+        filteredSection.removeAllRows()
+        guard isFiltered else {
+            tableView.reloadData()
+            return
+        }
+        
+        let section = tableData.sectionData(for: 0)
+        for i in 0..<section.rowCount() {
+            let row = section.getRow(i)
+            var groupMatched = false
+            if let widgetGroup = row.obj(forKey: "widget_group") as? WidgetGroup {
```

Comment:
if let widgetGroup = row.obj(forKey: "widget_group") as? WidgetGroup, widgetGroup.title.range(of: searchText, options: .caseInsensitive) != nil